realm/realm-library/src/main/java/io/realm/Realm.java

```diff
@@ -675,6 +687,11 @@ private Scanner getFullStringScanner(InputStream in) {
     public <E extends RealmModel> E createObject(Class<E> clazz) {
         checkIfValid();
         Table table = schema.getTable(clazz);
+        // Check and throw the exception earlier for a better exception message.
+        if (table.hasPrimaryKey()) {
+            throw new RealmException(String.format("'%s' has a primary key, use" +
+                    " 'createObject(Class<E>, Object)' to create it.", clazz.getSimpleName()));
```

Comment:
Model classes don't have `@Keep` annotation.
Table name is resolved by `RealmProxyMediator#getTableName()` and it uses `Class` itself, not it's name.